node-notes-cli/add.js

```diff
@@ -0,0 +1,15 @@
+import * as fs from 'node:fs/promises';
+import readNotes from './read.js';
+
+export default async function addNote(note) {
+  try {
+    const dataCopyObject = await readNotes(0);
+    const noteId = dataCopyObject.nextId.toString();
+    dataCopyObject.notes[noteId] = note.join(' ');
+    dataCopyObject.nextId++;
+    await fs.writeFile('data.json', JSON.stringify(dataCopyObject, null, 2));
```

Comment:
It's interesting that you encapsulate read into a separate file, but not write.

It would be good to encapsulate them both so you can easily change your mind about the implementation. For example, what if you decide to write binary instead of JSON, or put the notes into a database? How many places would you need to modify your code?